### PR TITLE
feat(File upload)

### PR DESCRIPTION
**Pull request by devvspaces**
- Created an endpoint for file uploads
- Endpoint returns the list of uploaded file URLs as response

image

### CODE REVIEW

File: backend/endpoints/files.py [hunk 1]
```diff
@@ -1,35 +1,76 @@
...
+from pydantic import AnyHttpUrl
+from schema.message import Message
+from schema.response import ResponseModel
+from starlette.responses import JSONResponse
+from utils.file_storage import FileStorage
 
+router = APIRouter()
 
-# @router.get("/org/{org_id}/rooms/{room_id}/files")
-# async def get_files(org_id: str, room_id: str, page: int = 1, limit: int = 15):
```
Comment (mikengr) [inline backend/endpoints/files.py:10]: @devvspaces is there a reason why the commented out endpoint was deleted?

File: backend/config/settings.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 from pydantic import BaseSettings
 
+
```
Comment (mikengr) [inline backend/config/settings.py]: @devvspaces please avoid touching files that are not related to the work you are doing. 8 files changes for this task gives me a lot of worry

File: backend/endpoints/files.py [hunk 3]
```diff
@@ -1,35 +1,76 @@
...
+    files = [
+        file.file for file in files
+    ]
+    response = await file_store.files_upload(files, token)
+
+    # Anything other than a list is an error
+    if not isinstance(response, list):
+        raise HTTPException(
+            status_code=status.HTTP_424_FAILED_DEPENDENCY,
+            detail={"Message not sent": response},
```
Comment (mikengr) [inline backend/endpoints/files.py]: this error checking is too ambiguous, not every error will be due to a failed dependency.

File: backend/endpoints/files.py [hunk 4]
```diff
@@ -1,35 +1,76 @@
...
+        )
+
+    # extract the urls from the response
+    file_urls: list[AnyHttpUrl] = [obj["file_url"] for obj in response]
+    message.files = file_urls
+
+    return JSONResponse(
+        content=ResponseModel.success(
+            data=message.dict(), message="Files uploaded successfully"),
+        status_code=status.HTTP_201_CREATED,
```
Comment (mikengr) [inline backend/endpoints/files.py:210]: but no message was created here, you just returned the empty message object, how will the frontend implement this?

**Review by mikengr — CHANGES_REQUESTED** [review]
good work @devvspaces 
Please remove these `/other.png` and `/files.txt`, I can't understand their purpose in this PR
I have also left some comments on some of your changes.

File: backend/endpoints/files.py (hunk 4 above)
Comment (devvspaces) [reply backend/endpoints/files.py:210]: @mikengr sir, they wanted just an upload endpoint, they said that's the way they wanted it, I had to do it for them.

**Comment by devvspaces** [comment]
Test Screenshots Below ( Automated and Manual Tests) @mikengr 

## Automated tests and server startup screen shot

test_1

## API Endpoint doc

test_2

## API Endpoint Manual Testing

test_3

**Review by mikengr — APPROVED** [review]
@devvspaces very good work, however putting your form validation in the same file as your endpoint is `anti-pattern`

In other to unblock others, I will merge this, but you will need to refactor `files.py`
